test/test_baza.rb

```diff
@@ -104,29 +106,47 @@ def test_push_with_big_meta
           ]
         )
       end
-    assert(req.join.include?('X-Zerocracy-Meta: '))
+    assert(req['x-zerocracy-meta'])
+  end
+
+  def test_push_compressed_content
+    req =
+      with_http_server(200, 'yes') do |baza|
+        baza.push('simple', 'hello, world!', %w[meta1 meta2 meta3])
+      end
+    assert_equal('application/zip', req.content_type)
+    assert_equal('gzip', req['content-encoding'])
+    body = Zlib::GzipReader.zcat(StringIO.new(req.body))
+    assert_equal('hello, world!', body)
+  end
+
+  def test_push_compression_disabled
+    req =
+      with_http_server(200, 'yes', compression: false) do |baza|
+        baza.push('simple', 'hello, world!', %w[meta1 meta2 meta3])
+      end
```

Comment:
@tank-bohr and here as well as above

Reply:
as well as above. it's not my decision. it's rubocop's